### PR TITLE
Python kubernetes key separator

### PR DESCRIPTION
**Pull request by bazsi**
This is a work in progress branch to find a good solution for representing kubernetes metadata that contains dots in JSON keys,
causing our name-value representation to become ambiguous

I don't feel the solution in this PR is right, but we can address this post 4.0, so I am splitting this off from #4173

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

File: modules/python-modules/syslogng/modules/kubernetes/scl/kubernetes.conf [hunk 1]
```diff
@@ -91,7 +92,7 @@ block parser kubernetes-metadata-parser (prefix() template("$FILE_NAME")) {
      };
 };
 
-block parser kubernetes-file-parser(prefix('.k8s.') cluster-name('k8s')) {
+block parser kubernetes-file-parser(prefix('.k8s.') cluster-name('k8s') key-delimiter() {
```
Comment (alltilla) [inline modules/python-modules/syslogng/modules/kubernetes/scl/kubernetes.conf:95]: A closing bracket is missing.
```suggestion
block parser kubernetes-file-parser(prefix('.k8s.') cluster-name('k8s') key-delimiter()) {
```

File: modules/python-modules/syslogng/modules/kubernetes/scl/kubernetes.conf [hunk 2]
```diff
@@ -115,7 +116,7 @@ block parser kubernetes-file-parser(prefix('.k8s.') cluster-name('k8s')) {
         timeout(10)
     );
 
-    kubernetes-metadata-parser(prefix(`prefix`));
+    kubernetes-metadata-parser(prefix(`prefix`) key-delimiter(`key-delimiter`);
```
Comment (alltilla) [inline modules/python-modules/syslogng/modules/kubernetes/scl/kubernetes.conf:119]: A closing bracket is missing.
```suggestion
    kubernetes-metadata-parser(prefix(`prefix`) key-delimiter(`key-delimiter`));
```

File: modules/python-modules/syslogng/modules/kubernetes/scl/kubernetes.conf [hunk 3]
```diff
@@ -136,6 +137,6 @@ block source kubernetes(prefix('.k8s.') base-dir("/var/log/containers") cluster-
                 flags(no-parse)
             );
         };
-        parser { kubernetes-file-parser(prefix(`prefix`) cluster-name(`cluster-name`)); };
+        parser { kubernetes-file-parser(prefix(`prefix`) cluster-name(`cluster-name`) key-delimiter(`key-delimiter`); };
```
Comment (alltilla) [inline modules/python-modules/syslogng/modules/kubernetes/scl/kubernetes.conf:140]: A closing bracket is missing.
```suggestion
        parser { kubernetes-file-parser(prefix(`prefix`) cluster-name(`cluster-name`) key-delimiter(`key-delimiter`)); };
```

**Review by alltilla — APPROVED** [review]

**Pull request merged**